src/dataflow-types/src/client.rs

```diff
@@ -1362,18 +1384,24 @@ where
                     Err(_) => clients_alive = false,
                 },
 
-                Some(res) = client.recv() => match res {
-                    Response::Storage(res) => {
+                response = client.recv() => match response {
+                    Ok(Some(Response::Storage(res))) => {
                         let _ = storage_response_tx.send(res);
                     }
-                    Response::Compute(res, instance) => match instances.get(&instance) {
+                    Ok(Some(Response::Compute(res, instance))) => match instances.get(&instance) {
                         None => {
                             error!("received response {res:?} for non-existent compute instance {instance:?}");
                         }
                         Some(response_tx) => {
                             let _ = response_tx.send(res);
                         }
                     },
+                    error @ Err(_) => {
+                        error.unwrap();
+                    },
+                    Ok(None) => {
+                        // Nothing, is what the prior code did.
```

Comment:
Yeah, wasn't sure what the protocol is here. Maybe breaking would be better. Not sure!

Reply:
Me neither! :D